File: Projetos/MVC Pesquisa_Ordenacao/mvc_java/ExemploMedicaoTempo.java
import java.util.List;
import java.util.ArrayList;

import model.Utilidades;
import view.Visao;

/*
Construir um programa que popule 2 listas: uma com números aleatórios e outra com números sequenciais... temporizando cada rotina
*/
/**
 * CONTROLLER: coordena o Model (Utilidades) e a View (Visao).
 */
public class ExemploMedicaoTempo {
    public static void main(String[] args) {
        long tempoInicio, tempoFim;
        List<Integer> listaAleatoria = new ArrayList<>();
        List<Integer> listaSequencial = new ArrayList<>();

        tempoInicio = System.nanoTime();
        // rotina1 - popular uma lista com 100000 aleatorios na faixa 100 a 100000
        Utilidades.popularLista(listaAleatoria, 100000, 100, 100000, true);
        // Visao.exibrLista(listaAleatoria, "Lista aleatoria");
        tempoFim = System.nanoTime();
        Visao.exibirTempo(1, (tempoFim - tempoInicio)/1000000);

        tempoInicio = System.nanoTime();
        // rotina2 - popular uma lista com 100000 de forma crescente 0 na posição 0, 1 na posição 1, e assim por diante
        Utilidades.popularLista(listaSequencial, 100000, 1, 100000, false);
        // Visao.exibrLista(listaSequencial, "Lista sequencial");
        tempoFim = System.nanoTime();
        Visao.exibirTempo(2, (tempoFim - tempoInicio)/1000000);
    }
}

File: Projetos/MVC Pesquisa_Ordenacao/mvc_python/exemplo_medicao_tempo.py
"""CONTROLLER: coordena o Model (utilidades) e a View (visao).

Construir um programa que popule 2 listas: uma com números aleatórios
e outra com números sequenciais... temporizando cada rotina.
"""
import time

from model import utilidades
from view import visao


def main():
    lista_aleatoria = []
    lista_sequencial = []

    tempo_inicio = time.perf_counter_ns()
    # rotina1 - popular uma lista com 100000 aleatorios na faixa 100 a 100000
    utilidades.popular_lista(lista_aleatoria, 100000, 100, 100000, True)
    # visao.exibr_lista(lista_aleatoria, "Lista aleatoria")
    tempo_fim = time.perf_counter_ns()
    visao.exibir_tempo(1, (tempo_fim - tempo_inicio) // 1000000)

    tempo_inicio = time.perf_counter_ns()
    # rotina2 - popular uma lista com 100000 de forma crescente
    utilidades.popular_lista(lista_sequencial, 100000, 1, 100000, False)
    # visao.exibr_lista(lista_sequencial, "Lista sequencial")
    tempo_fim = time.perf_counter_ns()
    visao.exibir_tempo(2, (tempo_fim - tempo_inicio) // 1000000)


if __name__ == "__main__":
    main()
